stock_shipment_management/model/stock_move.py

```diff
@@ -110,7 +110,7 @@ def _get_ship_partner_id(self):
     @api.multi
     def write(self, values):
         res = super(StockMove, self).write(values)
-        if values.get('state', '') == 'done':
+        if values.get('state', '') in ('done', 'cancel'):
```

Comment:
I'd argue you can leave out the second argument to `get()` but this is really a detail.